diffblue/cover-client-js
# Add default settings route and make analysis settings required [TG-10029, TG-10032]

### Context/purpose

The client should support the new `/default-settings` api route.

The client should understand that settings merging has been removed and that settings are now required to start an analysis.

### Implementation details

Add low level and OO methods for `getDefaultSettings`

Make `settings` a required parameter for the low level start analysis binding.

Settings remain optional for the OO `start` and `run` methods, and the default settings will be fetched and applied to the analysis if the settings are omitted.

Various type and docs updates, including removing references to `ignoreDefaults`.

### QA instructions

You should now be able to get default settings from the API.

You should be able to start an analysis with default settings by not providing settings when using `Analysis.run` or `Analysis.start`.

You should no longer be able to call the low level `startAnalysis` function without providing settings.

### Any unrelated changes?



### PR Checklist



- [x] I have given the PR a meaningful title which can be understood out of context (used in the changelog)
- [x] I have read through my code and fixed the indentation, followed naming conventions and removed leftover debugging
- [x] I have added the Jira ticket number(s) to the title with the format "Description of the feature [TG-123, TG-456]" and have copied the QA instructions to the Jira ticket (if one exists)
